src/metabase/upload/parsing.clj

```diff
@@ -35,7 +35,11 @@
   Supported formats:
     - yyyy-MM-dd"
   [s]
-  (t/local-date s))
```

Comment:
are you aiming for completeness here? if so we need to update `parse-datetime` too.

How about introducing a macro?

Reply:
and btw, this function can be made private. I suspect a bunch of others are too.

Reply:
> if so we need to update parse-datetime too.

Unlike `parse-date`, `parse-datetime` is not used in an `upload-type->parser` function, except to help implement `parse-as-datetime`. So it doesn't need the same exception handling.

> How about introducing a macro?

This is subjective but I don't think wrapping a body in a try/catch like this crosses my threshold for "so much boilerplate I want a macro to hide it"

> this function can be made private

Good point, I'll fix this and others